### PR TITLE
7530 skip rule validation

### PR DESCRIPTION
**Pull request by hooverlunch**

### CODE REVIEW

File: app/models/form.rb [hunk 1]
```diff
@@ -71,7 +71,6 @@ def self.receivable_association
   def self.terminate_sub_relationships(form_ids)
     Form.where(id: form_ids).update_all(original_id: nil)
     FormVersion.where(form_id: form_ids).delete_all
-    Questioning.where(form_id: form_ids).delete_all
```
Comment (meltheadorable) [inline app/models/form.rb]: Why this change?

File: app/models/mission.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Mission < ApplicationRecord
 
   scope(:sorted_by_name, -> { order("name") })
   scope(:sorted_recent_first, -> { order("missions.created_at DESC") })
-  scope(:for_user, ->(u) { where("missions.id IN 
+  scope(:for_user, ->(u) { where("missions.id IN
```
Comment (meltheadorable) [inline app/models/mission.rb]: Is there a way to write this query in ActiveRecord?

Something like `where(id: Assignment.where(user_id: u.id).pluck(:mission_id))` (this should pick up paranoia's default queries)?

I think we should be working to avoid hand-crafted SQL when possible since it helps us avoid a lot of the types of migration pains we experienced switching to  postgres and then to UUIDs.

File: db/migrate/20171219154111_convert_skip_rule_to_uuid.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class ConvertSkipRuleToUuid < ActiveRecord::Migration
```
Comment (meltheadorable) [inline db/migrate/20171219154111_convert_skip_rule_to_uuid.rb]: Good call. Do you know if there's a way to make UUID the new default on rails 4? Might be worth that config change if it can be done. Otherwise we should put something in the backlog to make sure we set that up on rails 5 as I'm pretty sure it _can_ be done there.

File: spec/models/skip_rule_spec.rb [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        rule = create(:skip_rule, source_item: qing2, conditions_attributes: [
+          {ref_qing_id: qing1.id, op: "eq", value: "5"},
+          {ref_qing_id: "", op: "", value: ""}
+        ])
+        expect(rule.conditions.count).to eq 1
+        expect(rule.conditions[0].ref_qing).to eq qing1
+      end
+    end
+
+    describe "skip_if" do
```
Comment (meltheadorable) [inline spec/models/skip_rule_spec.rb]: these specs look great 💯

File: spec/models/mission_destroy_spec.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  end
+  let(:deleted_counts) { expected_counts.map { |k, v| [k, 0] }.to_h }
+
+  it "should delete all objects in mission" do
+    expect(actual_counts).to eq expected_counts
+    mission.destroy
+    expect(actual_counts).to eq deleted_counts
+  end
+
+  def actual_counts
```
Comment (meltheadorable) [inline spec/models/mission_destroy_spec.rb]: Great refactor here.

File: app/models/form.rb (hunk 1 above)
Comment (hooverlunch) [reply app/models/form.rb]: Specs pass without it. Questionings get deleted by Mission directly. This code has a lot of quirks, just shaving off a few.

File: app/models/mission.rb (hunk 2 above)
Comment (hooverlunch) [reply app/models/mission.rb]: Sure I can try it. Or even `where(id: user.assignments.pluck(:id))`, right? 

This was just a whitespace change but I guess that's my rotten luck :P

File: db/migrate/20171219154111_convert_skip_rule_to_uuid.rb (hunk 3 above)
Comment (hooverlunch) [reply db/migrate/20171219154111_convert_skip_rule_to_uuid.rb]: http://www.mccartie.com/2015/10/20/default-uuid's-in-rails.html

I will add.

File: spec/models/mission_destroy_spec.rb (hunk 5 above)
Comment (hooverlunch) [reply spec/models/mission_destroy_spec.rb]: Thanks!

File: app/models/mission.rb (hunk 2 above)
Comment (hooverlunch) [reply app/models/mission.rb]: Actually turns out this method is a crappier version of user.missions and is only used once so I got rid of it :D
